lib/SILOptimizer/Transforms/DeadStoreElimination.cpp

```diff
@@ -117,6 +117,20 @@ findDeallocStackInst(AllocStackInst *ASI) {
   return DSIs;
 }
 
+/// Return the deallocate ref instructions corresponding to the given
+/// AllocRefInst.
+static llvm::SmallVector<SILInstruction *, 1>
+findDeallocRefInst(AllocRefInst *ARI) {
+  llvm::SmallVector<SILInstruction *, 1> DSIs;
+  for (auto UI = ARI->use_begin(), E = ARI->use_end(); UI != E; ++UI) {
+    if (auto *D = dyn_cast<DeallocRefInst>(UI->getUser())) {
+      if (D->isDeallocatingStack())
```

Comment:
This can be an assert. The corresponding dealloc_ref must have the [stack] flag set.